S/SIRIUS/build_tarballs.jl

```diff
@@ -0,0 +1,126 @@
+            -DCMAKE_INSTALL_PREFIX=$prefix \
+            -DCMAKE_BUILD_TYPE=Release \
+            -DBUILD_SHARED_LIBS=ON \
+            -DMPI_C_COMPILER=$bindir/mpicc \
+            -DMPI_CXX_COMPILER=$bindir/mpicxx"
+
+if [[ "${target}" == *-apple-mpich ]]; then
+  CMAKE_ARGS="${CMAKE_ARGS} \
+               -DMPI_C_LIB_NAMES='mpi;pmpi;hwloc' \
+               -DMPI_CXX_LIB_NAMES='mpicxx;mpi;pmpi;hwloc' \
+               -DMPI_mpicxx_LIBRARY=${libdir}/libmpicxx.dylib \
+               -DMPI_mpi_LIBRARY=${libdir}/libmpi.dylib \
+               -DMPI_pmpi_LIBRARY=${libdir}/libpmpi.dylib \
+               -DMPI_hwloc_LIBRARY=${libdir}/libhwloc.dylib"
+fi
+
+if [[ "${target}" == *-apple-mpitrampoline ]]; then
+  CMAKE_ARGS="${CMAKE_ARGS} \
+               -DMPI_C_LIB_NAMES='mpi;pmpi;hwloc' \
+               -DMPI_CXX_LIB_NAMES='mpicxx;mpi;pmpi;hwloc' \
+               -DMPI_mpicxx_LIBRARY=${libdir}/mpich/lib.libmpicxx.a \
+               -DMPI_mpi_LIBRARY=${libdir}/mpich/lib/libmpi.a \
+               -DMPI_pmpi_LIBRARY=${libdir}/mpich/lib/libpmpi.a \
+               -DMPI_hwloc_LIBRARY=${libdir}/libhwloc.dylib"
+fi
+
+#need to pass various results of CMake's try_run() for MPI compilers for succesful build
+CMAKE_ARGS="${CMAKE_ARGS} -DMPI_RUN_RESULT_C_libver_mpi_normal=0 \
+                          -DMPI_RUN_RESULT_C_libver_mpi_normal__TRYRUN_OUTPUT='' \
+                          -DMPI_RUN_RESULT_CXX_libver_mpi_normal=0 \
+                          -DMPI_RUN_RESULT_CXX_libver_mpi_normal__TRYRUN_OUTPUT='' \
+                          -DMPI_RUN_RESULT_Fortran_libver_mpi_F90_MODULE=0 \
+                          -DMPI_RUN_RESULT_Fortran_libver_mpi_F90_MODULE__TRYRUN_OUTPUT='' \
+                          -DMPI_RUN_RESULT_Fortran_libver_mpi_F08_MODULE=0 \
+                          -DMPI_RUN_RESULT_Fortran_libver_mpi_F08_MODULE__TRYRUN_OUTPUT=''"
+
+cmake .. ${CMAKE_ARGS}
+
+#On MacOS, need to explicitly remove the -fallow-argument-mismatch flag, because not recognized by Clang
+if [[ "${target}" == *-apple* ]]; then
+   cmake .. "-DMPI_Fortran_COMPILE_OPTIONS=''"
+fi
+
+make -j${nproc} install
+"""
+
+augment_platform_block = """
+    using Base.BinaryPlatforms
+    $(MPI.augment)
+    augment_platform!(platform::Platform) = augment_mpi!(platform)
+"""
+platforms = [Platform("x86_64", "linux"), Platform("aarch64", "linux"), Platform("aarch64", "macos")]
+filter!(p -> !(libc(p) == "musl"), platforms)
+platforms = expand_cxxstring_abis(platforms)
+
+platforms = expand_gfortran_versions(platforms)
+filter!(p -> !(libgfortran_version(p) < v"5"), platforms)
+
+products = [
+   LibraryProduct("libsirius", :libsirius)
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    Dependency("GSL_jll"), 
+    Dependency("pugixml_jll"), 
+    #Using either MKL or OPENBLAS32
+    Dependency("libblastrampoline_jll"),
+    Dependency("Libxc_jll"), 
+    Dependency("HDF5_jll"),
```

Comment:
```suggestion
    Dependency("HDF5_jll"; compat="~1.14.3"),
```

Reply:
Note that I did not make this change. Specifying the version of `HDF5_jll` to `1.14.3` means I also need to fix `MPItrampoline` to `5.3.1`. This leads to CMake stalling during the search for HDF5. I could either drop `MPItrampoline` support or accept an older version of `HDF5_jll`.

Reply:
Having a compat bound for hdf5 is not negotiable, it's necessary because we different ABIs in different versions